### ISSUE TITLE
PPC64:build failure ppc64/-m32 and bultin-atomics

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#2677]
@PHHargrove has found another problem we seem to be having with PPC64 with/without use of the --enable-builtin-atomics.

```
I have a standard Linux/ppc64 system with gcc-4.8.3
I have configured the 2.0.2rc3 tarball with
--prefix=... --enable-builtin-atomics \
CFLAGS=-m32 --with-wrapper-cflags=-m32 \
CXXFLAGS=-m32 --with-wrapper-cxxflags=-m32 \
FCFLAGS=-m32 --with-wrapper-fcflags=-m32 --disable-mpi-fortran
(Yes, I know the FCFLAGS are unnecessary).

I get a "make check" failure:
make[3]: Entering directory `/home/phargrov/OMPI/openmpi-2.0.2rc3-linux-ppc32-gcc/BLD/test/asm'
  CC       atomic_barrier.o
  CCLD     atomic_barrier
  CC       atomic_barrier_noinline-atomic_barrier_noinline.o
  CCLD     atomic_barrier_noinline
  CC       atomic_spinlock.o
  CCLD     atomic_spinlock
  CC       atomic_spinlock_noinline-atomic_spinlock_noinline.o
  CCLD     atomic_spinlock_noinline
  CC       atomic_math.o
  CCLD     atomic_math
atomic_math.o: In function `atomic_math_test':
atomic_math.c:(.text+0x78): undefined reference to `__sync_add_and_fetch_8'
collect2: error: ld returned 1 exit status
make[3]: *** [atomic_math] Error 1
```

The original posting on devel is at

https://www.mail-archive.com/devel@lists.open-mpi.org//msg19868.html

This issue is related to #2610 .  IN some cases we appear to be needing to use the --enable-builtin-atomics to work around bugs in the PPC64 atomics asm code in Open MPI and then in other cases it's necessary to use this feature to avoid hangs when running make (dist)check.

**Comment by hppritcha** [comment]
@jsquyres I propose making both of the PPC64 atomics blockers for 2.0.2.  I'll assign to IBM.

**Comment by jsquyres** [comment]
@gpaulsen @jjhursey Can you guys be prepared to talk about this one tomorrow on the webex?

**Comment by nysal** [comment]
@PHHargrove I'm trying to get access to a ppc64 system to reproduce this issue. I think the problem might be that when you enable builtin atomics OPAL_HAVE_ATOMIC_MATH_64 is unconditionally defined as 1. Not sure if 64-bit atomics are supported on a 32-bit host for all architectures.

**Comment by hppritcha** [comment]
looks like there's a fix on master.  @hjelmn cherry-pick over from master to v2.0.x

**Comment by jsquyres** [comment]
@hjelmn cites d99a9786b6a14dd09b7bb2c3c52823fea4f6ce07 on master

**Comment by jjhursey** [comment]
Ref PR #2698

**Comment by hppritcha** [comment]
closed via #2698
